src/components/DappBrowser/search-input/SearchInput.tsx

```diff
@@ -74,85 +75,104 @@ const TheeDotMenu = function TheeDotMenu({
     }
   }, [addFavorite, formattedUrlValue, isFavorite, removeFavorite]);
 
-  const menuConfig = useMemo(
-    () => ({
-      menuTitle: '',
-      menuItems: [
-        {
-          actionKey: 'share',
-          actionTitle: 'Share',
-          icon: {
-            iconType: 'SYSTEM',
-            iconValue: 'square.and.arrow.up',
-          },
+  const menuConfig = useMemo(() => {
+    const menuItems = [
+      {
+        actionKey: 'share',
+        actionTitle: 'Share',
+        icon: {
+          iconType: 'SYSTEM',
+          iconValue: 'square.and.arrow.up',
         },
-        !activeTabInfo.value.isGoogleSearch
-          ? {
-              actionKey: 'favorite',
-              actionTitle: isFavorite ? 'Undo Favorite' : 'Favorite',
-              icon: {
-                iconType: 'SYSTEM',
-                iconValue: isFavorite ? 'star.slash' : 'star',
-              },
-            }
-          : {},
-        canGoForward
-          ? {
-              actionKey: 'forward',
-              actionTitle: 'Forward',
-              icon: {
-                iconType: 'SYSTEM',
-                iconValue: 'arrowshape.forward',
-              },
-            }
-          : {},
-        canGoBack
-          ? {
-              actionKey: 'back',
-              actionTitle: 'Back',
-              icon: {
-                iconType: 'SYSTEM',
-                iconValue: 'arrowshape.backward',
-              },
-            }
-          : {},
-      ],
-    }),
+      },
+    ];
+
+    const isGoogleSearch = tabUrl?.includes('google.com/search');
+    if (!isGoogleSearch) {
+      menuItems.push({
+        actionKey: 'favorite',
+        actionTitle: isFavorite ? i18n.t(i18n.l.dapp_browser.menus.undo_favorite) : i18n.t(i18n.l.dapp_browser.menus.favorite),
+        icon: {
+          iconType: 'SYSTEM',
+          iconValue: isFavorite ? 'star.slash' : 'star',
+        },
+      });
+    }
+    if (canGoForward) {
+      menuItems.push({
+        actionKey: 'forward',
+        actionTitle: i18n.t(i18n.l.dapp_browser.menus.forward),
+        icon: {
+          iconType: 'SYSTEM',
+          iconValue: 'arrowshape.forward',
+        },
+      });
+    }
+    if (canGoBack) {
+      menuItems.push({
+        actionKey: 'back',
+        actionTitle: i18n.t(i18n.l.dapp_browser.menus.back),
+        icon: {
+          iconType: 'SYSTEM',
+          iconValue: 'arrowshape.backward',
+        },
+      });
+    }
 
-    [activeTabInfo, canGoBack, canGoForward, isFavorite]
+    return {
+      menuTitle: '',
+      menuItems,
+    };
+  }, [canGoBack, canGoForward, isFavorite, tabUrl]);
+
+  const onPressMenuItem = useCallback(
+    async ({ nativeEvent: { actionKey } }: { nativeEvent: { actionKey: 'share' | 'favorite' | 'back' | 'forward' } }) => {
```

Comment:
would prefer these all to be typed properly going forward.

import { OnPressMenuItemEventObject } from 'react-native-ios-context-menu';